<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Easily Create Interactive Oncoplots • ggoncoplot</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Easily Create Interactive Oncoplots">
<meta name="description" content="Generate oncoplots from tabular mutational data. Optionally make these oncoplots interactive, with a fully customisable tooltip.">
<meta property="og:description" content="Generate oncoplots from tabular mutational data. Optionally make these oncoplots interactive, with a fully customisable tooltip.">
<meta property="og:image" content="https://selkamand.github.io/ggoncoplot/logo.png">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">ggoncoplot</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/essential_resources.html">Essential Resources</a></li>
    <li><a class="dropdown-item" href="articles/manual.html">Manual</a></li>
    <li><a class="dropdown-item" href="articles/sorting_algorithm.html">Sorting Algorithm</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/selkamand/ggoncoplot/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="ggoncoplot-">ggoncoplot 
<a class="anchor" aria-label="anchor" href="#ggoncoplot-"></a>
</h1>
</div>
<!-- badges: start -->

<!-- badges: end -->
<p>The <strong>ggoncoplot</strong> R package generates interactive oncoplots to visualize mutational patterns across patient cancer cohorts.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install the ggoncoplot package (from R-universe) run:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'ggoncoplot'</span>, repos <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'https://selkamand.r-universe.dev'</span>, <span class="st">'https://cloud.r-project.org'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Alternatively, install the development version of ggoncoplot from github:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install remotes package (required to install from github)</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"remotes"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Install ggoncoplot</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">'selkamand/ggoncoplot'</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<p>For complete usage, see the ggoncoplot <a href="https://selkamand.github.io/ggoncoplot/articles/manual.html">manual</a>.</p>
<div class="section level3">
<h3 id="input">Input<a class="anchor" aria-label="anchor" href="#input"></a>
</h3>
<p>The input for ggoncoplot is a <strong>data.frame</strong>/<strong>data.table</strong>/<strong>tibble</strong> with 1 row per mutation in cohort and columns describing the following:</p>
<ul>
<li><p>Gene Symbol</p></li>
<li><p>Sample Identifier</p></li>
<li><p>(optional) mutation type</p></li>
<li><p>(optional) tooltip (character string: what we show on mouse hover over a particular mutation)</p></li>
</ul>
<p>These columns can be in any order, and named anything. You define the mapping of your input dataset columns to the required features in the call to <strong>ggoncoplot</strong>.</p>
</div>
<div class="section level3">
<h3 id="basic-example">Basic Example<a class="anchor" aria-label="anchor" href="#basic-example"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/selkamand/ggoncoplot" class="external-link">ggoncoplot</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># TCGA GBM dataset from TCGAmuations package</span></span>
<span><span class="va">gbm_csv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">'ggoncoplot'</span>, <span class="st">"testdata/GBM_tcgamutations_mc3_maf.csv.gz"</span><span class="op">)</span></span>
<span><span class="va">gbm_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">gbm_csv</span>, header<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gbm_df</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="reference/ggoncoplot.html">ggoncoplot</a></span><span class="op">(</span></span>
<span>    col_genes <span class="op">=</span> <span class="st">'Hugo_Symbol'</span>, </span>
<span>    col_samples <span class="op">=</span> <span class="st">'Tumor_Sample_Barcode'</span>, </span>
<span>    col_mutation_type <span class="op">=</span> <span class="st">'Variant_Classification'</span>, </span>
<span>    topn <span class="op">=</span> <span class="fl">10</span>, </span>
<span>    interactive <span class="op">=</span> <span class="cn">FALSE</span> <span class="co"># Set to `TRUE` to enable tooltips &amp; cross-linking</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Identify Class ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ℹ Found 7 unique mutation types in input set</span></span>
<span><span class="co">#&gt; ℹ 0/7 mutation types were valid PAVE terms</span></span>
<span><span class="co">#&gt; ℹ 0/7 mutation types were valid SO terms</span></span>
<span><span class="co">#&gt; ℹ 7/7 mutation types were valid MAF terms</span></span>
<span><span class="co">#&gt; ✔ Mutation Types are described using valid MAF terms ... using MAF palete</span></span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-2-1.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="making-oncoplots-interactive">Making oncoplots interactive<a class="anchor" aria-label="anchor" href="#making-oncoplots-interactive"></a>
</h3>
<p>To turn on interactive features (tooltips, data-linking, etc), set the argument <code>interactive=TRUE</code>. See the <a href="https://selkamand.github.io/ggoncoplot/articles/manual.html">manual</a> for examples of interactive oncoplots, including how to set up data-crosslinking (shown below).</p>
<p><img src="reference/figures/interactive_oncoplot.gif"></p>
</div>
<div class="section level3">
<h3 id="add-marginal-plots">Add marginal plots<a class="anchor" aria-label="anchor" href="#add-marginal-plots"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gbm_df</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="reference/ggoncoplot.html">ggoncoplot</a></span><span class="op">(</span></span>
<span>    col_genes <span class="op">=</span> <span class="st">'Hugo_Symbol'</span>, </span>
<span>    col_samples <span class="op">=</span> <span class="st">'Tumor_Sample_Barcode'</span>, </span>
<span>    col_mutation_type <span class="op">=</span> <span class="st">'Variant_Classification'</span>, </span>
<span>    topn <span class="op">=</span> <span class="fl">10</span>, </span>
<span>    draw_gene_barplot <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>    draw_tmb_barplot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    interactive <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Identify Class ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ℹ Found 7 unique mutation types in input set</span></span>
<span><span class="co">#&gt; ℹ 0/7 mutation types were valid PAVE terms</span></span>
<span><span class="co">#&gt; ℹ 0/7 mutation types were valid SO terms</span></span>
<span><span class="co">#&gt; ℹ 7/7 mutation types were valid MAF terms</span></span>
<span><span class="co">#&gt; ✔ Mutation Types are described using valid MAF terms ... using MAF palete</span></span>
<span><span class="co">#&gt; ! TMB plot: Ignoring `col_mutation_type` since `log10_transform = TRUE`.</span></span>
<span><span class="co">#&gt; This is because you cannot accurately plot stacked bars on a logarithmic scale</span></span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-3-1.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="add-clinical-metadata">Add clinical metadata<a class="anchor" aria-label="anchor" href="#add-clinical-metadata"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gbm_clinical_csv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"ggoncoplot"</span>, <span class="st">"testdata/GBM_tcgamutations_mc3_clinical.csv"</span><span class="op">)</span></span>
<span><span class="va">gbm_clinical_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">gbm_clinical_csv</span>, header <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gbm_df</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="reference/ggoncoplot.html">ggoncoplot</a></span><span class="op">(</span></span>
<span>   col_genes <span class="op">=</span> <span class="st">"Hugo_Symbol"</span>,</span>
<span>   col_samples <span class="op">=</span> <span class="st">"Tumor_Sample_Barcode"</span>,</span>
<span>   col_mutation_type <span class="op">=</span> <span class="st">"Variant_Classification"</span>,</span>
<span>   metadata <span class="op">=</span> <span class="va">gbm_clinical_df</span>,</span>
<span>   cols_to_plot_metadata <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'gender'</span>, <span class="st">'histological_type'</span>, <span class="st">'prior_glioma'</span>, <span class="st">'tumor_tissue_site'</span><span class="op">)</span>,</span>
<span>   draw_tmb_barplot <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>   draw_gene_barplot <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>   show_all_samples <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>   interactive <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co">#&gt; ℹ 2 samples with metadata have no mutations. Fitering these out</span></span>
<span><span class="co">#&gt; ℹ To keep these samples, set `metadata_require_mutations = FALSE`. To view them in the oncoplot ensure you additionally set `show_all_samples = TRUE`</span></span>
<span><span class="co">#&gt; → TCGA-06-0165-01</span></span>
<span><span class="co">#&gt; → TCGA-06-0167-01</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Identify Class ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ℹ Found 7 unique mutation types in input set</span></span>
<span><span class="co">#&gt; ℹ 0/7 mutation types were valid PAVE terms</span></span>
<span><span class="co">#&gt; ℹ 0/7 mutation types were valid SO terms</span></span>
<span><span class="co">#&gt; ℹ 7/7 mutation types were valid MAF terms</span></span>
<span><span class="co">#&gt; ✔ Mutation Types are described using valid MAF terms ... using MAF palete</span></span>
<span><span class="co">#&gt; ! TMB plot: Ignoring `col_mutation_type` since `log10_transform = TRUE`.</span></span>
<span><span class="co">#&gt; This is because you cannot accurately plot stacked bars on a logarithmic scale</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Plotting Sample Metadata ────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Sorting</span></span>
<span><span class="co">#&gt; ℹ Sorting X axis by: Order of appearance</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Generating Plot</span></span>
<span><span class="co">#&gt; ℹ Found 4 plottable columns in data</span></span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-4-1.png" width="100%"></p>
</div>
</div>
<div class="section level2">
<h2 id="statement-of-need">Statement of Need<a class="anchor" aria-label="anchor" href="#statement-of-need"></a>
</h2>
<p>Oncoplots are highly effective for visualising mutation data in cancer cohorts but are challenging to generate with the major R plotting systems (base, lattice, or ggplot2) due to their algorithmic and graphical complexity. Simplifying the process of generating oncoplots would make them more accessible to researchers. Existing packages including <a href="https://github.com/jokergoo/ComplexHeatmap" class="external-link">ComplexHeatmap</a>, <a href="https://github.com/PoisonAlien/maftools" class="external-link">maftools</a>, and <a href="https://github.com/griffithlab/GenVisR" class="external-link">genVisR</a> all make static oncoplots easier to create, but there is still a significant unmet need for a user-friendly method of creating oncoplots with the following features:</p>
<ul>
<li><p><strong>Interactive plots</strong>: Customizable tooltips, cross-selection of samples across different plots, and auto-copying of sample identifiers on click. This enables exploration of multiomic datasets.</p></li>
<li><p><strong>Support for tidy datasets</strong>: Compatibility with tidy, tabular mutation-level formats that cancer cohort datasets are typically stored in. This greatly improves the range of datasets that can be quickly and easily visualised in an oncoplot since genomic data in Mutation Annotation Format (MAF) files and relational databases usually follow this structure.</p></li>
<li><p><strong>Auto-colouring</strong>: Automatic selection of accessible colour palettes for datasets where the consequence annotations are aligned with standard variant effect dictionaries including <strong>P</strong>rediction and <strong>A</strong>nnotation of <strong>V</strong>ariant <strong>E</strong>ffects (PAVE), <strong>S</strong>equence <strong>O</strong>ntology (SO) and MAF Variant Classifications.</p></li>
<li><p><strong>Versatility</strong>: The ability to visualize entities other than gene mutations, such as noncoding features (e.g., promoter or enhancer mutations) and non-genomic entities (e.g., microbial presence in microbiome datasets).</p></li>
</ul>
<p>We developed ggoncoplot as the first R package to address all these challenges together (). Examples of all key features are available in the <a href="https://selkamand.github.io/ggoncoplot/articles/manual.html">ggoncoplot manual</a>.</p>
<p>A full comparison of ggoncoplot features with similar tools is available <a href="paper/ggoncoplot_comparison.pdf">here</a></p>
</div>
<div class="section level2">
<h2 id="scalability">Scalability<a class="anchor" aria-label="anchor" href="#scalability"></a>
</h2>
<p>ggOncoplot can produce its default, interactive oncoplot on even the largest TCGA cancer cohort (Breast Cancer - BRCA) which contains 1026 samples in 0.81 seconds (Macbook Pro; M3 Pro Chip; 18GB ram). Since the number of mutations in a genomic dataset does not change the number of tiles rendered in the final oncoplot, a 10x increase in variant number (1,350,300) takes only 1.3x longer to plot.</p>
<p>Code used for benchmarking is shown below.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/joshuaulrich/microbenchmark/" class="external-link">microbenchmark</a></span><span class="op">)</span> <span class="co"># install.packages("microbenchmark")</span></span>
<span></span>
<span><span class="co"># Setup Data</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"testdata/BRCA_tcgamutations_mc3.csv.gz"</span>, package <span class="op">=</span> <span class="st">"ggoncoplot"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Increase variant count by 10x</span></span>
<span><span class="va">data_10x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="st">"rbind"</span>, <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">replicate</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">10</span>, <span class="va">data</span>, simplify <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Benchmark</span></span>
<span><span class="fu">microbenchmark</span><span class="op">(</span></span>
<span>  interactive <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="reference/ggoncoplot.html">ggoncoplot</a></span><span class="op">(</span><span class="va">data</span>, col_samples <span class="op">=</span> <span class="st">"Sample"</span>, col_genes <span class="op">=</span> <span class="st">"Gene"</span>, col_mutation_type <span class="op">=</span> <span class="st">"MutationType"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>, </span>
<span>  interactive_10x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="reference/ggoncoplot.html">ggoncoplot</a></span><span class="op">(</span><span class="va">data_10x</span>, col_samples <span class="op">=</span> <span class="st">"Sample"</span>, col_genes <span class="op">=</span> <span class="st">"Gene"</span>, col_mutation_type <span class="op">=</span> <span class="st">"MutationType"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  static <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="reference/ggoncoplot.html">ggoncoplot</a></span><span class="op">(</span><span class="va">data</span>, col_samples <span class="op">=</span> <span class="st">"Sample"</span>, col_genes <span class="op">=</span> <span class="st">"Gene"</span>, col_mutation_type <span class="op">=</span> <span class="st">"MutationType"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>, interactive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  static_10x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="reference/ggoncoplot.html">ggoncoplot</a></span><span class="op">(</span><span class="va">data_10x</span>, col_samples <span class="op">=</span> <span class="st">"Sample"</span>, col_genes <span class="op">=</span> <span class="st">"Gene"</span>, col_mutation_type <span class="op">=</span> <span class="st">"MutationType"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>, interactive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  times <span class="op">=</span> <span class="fl">18</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="limitations">Limitations<a class="anchor" aria-label="anchor" href="#limitations"></a>
</h2>
<p>Responsiveness of interactive graphics may slow as the number of tiles in oncoplot (samples x genes increases) especially when tooltips contain large amounts of information. On a MacBook Pro (M3 Pro chip with 18GB of memory) the largest TCGA dataset (BRCA) including 1026 samples can be rendered with sample IDs in tooltip with no noticeable delay in tooltip responsiveness.</p>
</div>
<div class="section level2">
<h2 id="acknowledgements">Acknowledgements<a class="anchor" aria-label="anchor" href="#acknowledgements"></a>
</h2>
<p>We acknowledge the developers and contributors whose packages and efforts were integral to the development of ggoncoplot:</p>
<ul>
<li>
<strong>David Gohel</strong> for the <code>ggiraph</code> package, which enables the interactivity of ggoncoplot.</li>
<li>
<strong>Thomas Lin Pedersen</strong> for his contributions to the <code>patchwork</code> package and the maintenance of <code>ggplot2</code>.</li>
<li>
<strong>Hadley Wickham</strong> and all contributors to the <code>ggplot2</code> package, which provides a robust foundation for data visualization in R.</li>
</ul>
<p>Additionally, we thank <strong>Dr. Marion Mateos</strong> for her insightful feedback during the early stages of ggoncoplot development.</p>
<p>We also acknowledge Cerami <em>et al.</em> for their early, pioneering development of interactive oncoplots written in javascript, published in <a href="https://doi.org/10.1158/2159-8290.CD-12-0095" class="external-link">Cancer Discovery in 2012</a> and accessible as a web-app from <a href="https://www.cbioportal.org/oncoprinter" class="external-link">cbioportal</a>.</p>
</div>
<div class="section level2">
<h2 id="community-contributions">Community Contributions<a class="anchor" aria-label="anchor" href="#community-contributions"></a>
</h2>
<p>All types of contributions are encouraged and valued. See our <a href="CONTRIBUTING.html">guide to community contributions</a> for different ways to help.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/selkamand/ggoncoplot/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/selkamand/ggoncoplot/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CONTRIBUTING.html">Contributing guide</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing ggoncoplot</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Sam El-Kamand <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-2270-8088" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental" class="external-link"><img src="https://img.shields.io/badge/lifecycle-experimental-orange.svg" alt="Lifecycle: experimental"></a></li>
<li><a href="https://CRAN.R-project.org/package=ggoncoplot" class="external-link"><img src="https://www.r-pkg.org/badges/version/ggoncoplot" alt="CRAN status"></a></li>
<li><a href="https://app.codecov.io/gh/selkamand/ggoncoplot?branch=master" class="external-link"><img src="https://codecov.io/gh/selkamand/ggoncoplot/branch/master/graph/badge.svg" alt="Codecov test coverage"></a></li>
<li><a href="https://github.com/selkamand/ggoncoplot/actions/workflows/R-CMD-check.yaml" class="external-link"><img src="https://github.com/selkamand/ggoncoplot/actions/workflows/R-CMD-check.yaml/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://github.com/selkamand/ggoncoplot" class="external-link"><img src="https://img.shields.io/github/languages/code-size/selkamand/ggoncoplot.svg"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Sam El-Kamand.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
